### PR TITLE
Enhance Xtext Junit Deprecations

### PR DESCRIPTION
**Pull request by cdietrich**

### CODE REVIEW

File: org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractXtextTests.java [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+import com.google.inject.Provider;
+import com.google.inject.TypeLiteral;
+import com.google.inject.name.Names;
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ * @since 2.30
+ */
+@Beta
+public abstract class AbstractXtextTests extends Assert implements ResourceLoadHelper {
```
Comment (cdietrich) [inline org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractXtextTests.java:72]: am not sure if we want this as api. @szarnekow what do you think? copy to the tests instead?

Comment (cdietrich) [reply org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractXtextTests.java:72]: can you also check if i copy&pasted the correct variant

**Comment by cdietrich** [comment]
see also
https://github.com/eclipse/xtext/issues/2124
https://github.com/eclipse/xtext-core/issues/2029

File: org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractXtextTests.java (hunk 1 above)
Comment (szarnekow) [reply org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractXtextTests.java:72]: I don't think we've used this base class in recent test cases. I think it's convoluted and was superseded by the `@InjectWith` / `@RegisterExtension`

Comment (szarnekow) [reply org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractXtextTests.java:72]: Yes, I think so. C&P& strip down in the test projects.

Comment (cdietrich) [reply org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractXtextTests.java:72]: any hints to strip

**Comment by cdietrich** [comment]
@szarnekow ok to merge?

File: org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractXtextTests.java (hunk 1 above)
Comment (szarnekow) [reply org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractXtextTests.java]: Subsequent work

**Pull request merged**